src/librustc/session/config.rs

```diff
@@ -1447,7 +1447,7 @@ options! {DebuggingOptions, DebuggingSetter, basic_debugging_options,
         "don't interleave execution of lints; allows benchmarking individual lints"),
     crate_attr: Vec<String> = (Vec::new(), parse_string_push, [TRACKED],
         "inject the given attribute in the crate"),
-    self_profile: bool = (false, parse_bool, [UNTRACKED],
+    self_profile: PgoGenerate = (PgoGenerate::Disabled, parse_pgo_generate, [UNTRACKED],
```

Comment:
Can we rename `PgoGenerate` to something more general? It's a little weird IMO to see it used for non-PGO options. Perhaps something like `OptionalDirectory`?

Reply:
Yeah, I know `:)` 

The whole compiler flag parsing infrastructure took a wrong turn somewhere. There already is a `parse_opt_path` function but it's not what you'd think. I think the inconsistencies come from sometimes explicitly modelling missing flags as `None` and sometimes as a default value.

I'm having trouble coming up with a proper names since there are actually two levels of "optionality" here. `SwitchWithOptPath`? `ResultWithOptPath`? `OptOptPath`?

Reply:
👍 for `SwitchWithOptPath`